src/vs/workbench/browser/parts/editor/editorCommands.ts

```diff
@@ -1338,6 +1341,27 @@ function registerOtherEditorCommands(): void {
 		}
 	});
 
+	KeybindingsRegistry.registerCommandAndKeybindingRule({
+		id: OPEN_FILE_IN_EDITOR,
+		weight: KeybindingWeight.WorkbenchContrib,
+		when: inDiffEditor,
+		primary: KeyChord(KeyMod.CtrlCmd | KeyCode.KeyK, KeyMod.Shift | KeyCode.KeyO),
+		handler: async (accessor, resourceOrContext?: URI | IEditorCommandsContext, context?: IEditorCommandsContext) => {
+			let resource: URI | undefined;
+			const editorService = accessor.get(IEditorService);
+			const originalEditor = (editorService.activeTextEditorControl as IDiffEditor).getOriginalEditor();
+			if (originalEditor.hasTextFocus()) {
+				resource = (editorService.activeEditor as DiffEditorInput).original.resource;
+			} else {
+				// Default back to modified editor
+				resource = (editorService.activeEditor as DiffEditorInput).modified.resource;
+			}
+			editorService.openEditor({
+				resource,
```

Comment:
I do not think this is robust. You assume that every editor can be opened based on its `resource` but that may not work for all editors. A safer approach would be to reach into the `DiffEditorInput` and obtain the inputs of the original and modified side and open that instead.

Reply:
I now pass in the `DiffEditorInput` instead of the resource, although it does complain this way is deprecated.